src/api/apollo.js

```diff
@@ -1,48 +1,150 @@
 /* eslint-disable no-underscore-dangle */
-import ApolloClient from 'apollo-client';
-import { ApolloLink } from 'apollo-link';
-import {
-	IntrospectionFragmentMatcher,
-	InMemoryCache,
-	defaultDataIdFromObject,
-} from 'apollo-cache-inmemory';
+import { ApolloLink, ApolloClient, InMemoryCache } from '@apollo/client/core';
+
+import loanResolverFactory from '@/api/localResolvers/loan';
 import Auth0LinkCreator from './Auth0Link';
 import BasketLinkCreator from './BasketLink';
 import ContentfulPreviewLink from './ContentfulPreviewLink';
 import ExperimentIdLink from './ExperimentIdLink';
 import HttpLinkCreator from './HttpLink';
 import NetworkErrorLink from './NetworkErrorLink';
 import SnowplowSessionLink from './SnowplowSessionLink';
-import initState from './localState';
+// import initState from './localState';
+
+const { resolvers } = loanResolverFactory();
+const { LoanPartner, LoanDirect } = resolvers;
+
+export const typePolicies = {
+	activeLoan: {
+		merge: true
+	},
+	LoanChannel: {
+		keyFields: ['id'],
+	},
+	RecLoanChannel: {
+		keyFields: ['id'],
+	},
+	LoanPartner: {
+		fundraisingPercent: {
+			read(_, { readField }) {
+				return LoanPartner.fundraisingPercent(readField('loan'));
+			}
+		},
+		fundraisingTimeLeft: {
+			read(_, { readField }) {
+				return LoanPartner.fundraisingTimeLeft(readField('loan'));
+			}
+		},
+		unreservedAmount: {
+			read(_, { readField }) {
+				return LoanPartner.unreservedAmount(readField('loan'));
+			}
+		},
+		fundraisingTimeLeftMilliseconds: {
+			read(_, { readField }) {
+				return LoanPartner.fundraisingTimeLeftMilliseconds(readField('loan'));
+			}
+		},
+	},
+	LoanDirect: {
+		fundraisingPercent: {
+			read(_, { readField }) {
+				return LoanDirect.fundraisingPercent(readField('loan'));
+			}
+		},
+		fundraisingTimeLeft: {
+			read(_, { readField }) {
+				return LoanDirect.fundraisingTimeLeft(readField('loan'));
+			}
+		},
+		unreservedAmount: {
+			read(_, { readField }) {
+				return LoanDirect.unreservedAmount(readField('loan'));
+			}
+		},
+		fundraisingTimeLeftMilliseconds: {
+			read(_, { readField }) {
+				return LoanDirect.fundraisingTimeLeftMilliseconds(readField('loan'));
+			}
+		},
+	},
+	loanSearchState: {
+		keyFields: ['id'],
+	},
+	Manifest: {
+		keyFields: ['id'],
+	},
+	shop: {
+		keyFields: ['id'],
+	},
+	Setting: {
+		keyFields: ['key'],
+	},
+	Query: {
+		fields: {
+			community: {
+				merge: true
+			},
+			general: {
+				merge: true
+			},
+			// The ml field doesn't have an ID
+			ml: {
+				merge: true,
+			},
+			// The contentful field doesn't have an ID
+			contentful: {
+				merge: true,
+			},
+			// The lend field doesn't have an ID
+			lend: {
+				merge: true,
+			},
+			// The my field doesn't have an ID
```

Comment:
You could combine this identical comments (and some are missing on fields here) to one comment above line 84 (fields) to explain why we use merge below.